patch(style): Navbar UI

dipanshurdev opened the pull request:
## Description

Redesigned the header component with improved UI/UX, GitHub stars (26.8k+) integration, and optimized caching strategy.

Key Changes:
UI/UX Improvements:

- Redesigned navigation with right-aligned, clean text links (Blog, Contributors)
- Added GitHub stars badge with real-time data from GitHub API
- Enhanced mobile menu with improved responsive design
- Maintained Projects button as primary CTA with original styling
- Added smooth hover effects and transitions throughout

Technical Enhancements:

- Implemented Next.js server-side caching for GitHub API calls (1-hour cache)
- Created custom useGitHubStars hook with client-side caching (5-minute cache)
- Added proper loading states and error handling with fallback values
- Optimized API calls to prevent rate limiting and improve performance

Performance Benefits

- Server-first data fetching reduces client-side loading
- Multi-layer caching (Next.js fetch cache + custom hook cache)
- Rate limit protection prevents API abuse
- Graceful fallbacks ensure stars always display

Mobile Responsiveness

- Clean mobile menu overlay with backdrop blur
- Touch-friendly navigation elements
- Consistent styling across all screen sizes

## How Has This Been Tested?

localhost

## Screenshots (if applicable)

### Before (old)

<img width="1879" height="173" alt="Screenshot 2025-07-24 114000" src="https://github.com/user-attachments/assets/cca4d87e-35d0-4528-a230-fd0f3ecf39bf" />

### After (updated)

<img width="1898" height="184" alt="Screenshot 2025-07-24 155342" src="https://github.com/user-attachments/assets/944a4669-3dfa-4107-aad9-0e28b37d449e" />
14-2b35-43c2-8d4a-7fa8f0a33015" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

* **New Features**
  * Added a custom hook to manage fetching and caching GitHub star counts with improved loading and error states.

* **Refactor**
  * Modularized and improved the GitHub stars fetching logic, adding server-side caching and better error handling.

* **Bug Fixes**
  * Improved error handling and fallback for GitHub API rate limits or failures.

* **Chores**
  * Removed GitHub star count and related asynchronous logic from the footer for a cleaner, static display.
  * Disabled environment variable usage and related dynamic script attributes.
  * Commented out environment configuration and authentication keys setup.

* **UI Improvements**
  * Limited displayed projects to the first five on the projects page for concise presentation.
  * Enabled drag-and-drop functionality for text elements in the timeline editor.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mezotv reviewed (changes requested):
Please just remove the code instead of commenting it out, also please keep the current border radius on the cta button in the navbar.

dipanshurdev commented:
> Please just remove the code instead of commenting it out, also please keep the current border radius on the cta button in the navbar.

Done, removed commented code and fixed the border radius.

<img width="1898" height="184" alt="Screenshot 2025-07-24 155342" src="https://github.com/user-attachments/assets/f8904020-6992-4629-8aa0-f61ee12fa657" />

mezotv commented on apps/web/src/components/header.tsx:
```diff
@@ -71,7 +71,7 @@ export function Header({ initialStars }: HeaderProps) {
                   <div className="w-8 h-4 bg-gray-200 dark:bg-gray-600 rounded animate-pulse"></div>
                 ) : (
-                  `${stars}+`
+                  `${isLoading ? "28k" : stars}+`
```
You hardcode this here but in useGhStars you create a variable for it, would you mind moving these to a constant.ts file/folder and import them from there? So when we ever want to change the number it changed everywhere.

dipanshurdev replied:
fixed, created a separate file inside the constants folder, and moved the variables into it.

mezotv commented on apps/web/src/hooks/useGhStars.ts:
```diff
@@ -9,88 +9,130 @@ interface GitHubStarsState {
 const CLIENT_CACHE_DURATION = 5 * 60 * 1000;
+const MAX_POLL_RETRIES = 50;
+const POLL_TIMEOUT = 10000;
+const FALLBACK_STARS = "28k";
```
Same thing as above with the constant.ts file

mezotv commented on apps/web/src/lib/fetch-github-stars.ts:
```diff
@@ -6,6 +6,10 @@ interface GitHubRepoData {
   stargazers_count: number;
 }
 
+// Constants for maintainability
+const FALLBACK_STARS = "28k";
+const CACHE_DURATION_SECONDS = 3600; // 1 hour
+
```
Same thing as above

mezotv commented:
I just checked out the styling locally and on mid sized devices here is no margin on the left/right which looks bad.
<img width="894" height="140" alt="image" src="https://github.com/user-attachments/assets/24eb7b12-75e2-4297-a144-49cd0764aded" />


Also the colors and styling of the mobile menu does not fit our colors
<img width="347" height="290" alt="image" src="https://github.com/user-attachments/assets/79f38efd-3491-4bdd-b8dd-92b42c63820a" />

dipanshurdev reviewed (commented):
done
